synapse/util/async_helpers.py

```diff
@@ -525,28 +527,11 @@ def failure_cb(val):
     return new_d
 
 
-@attr.s(slots=True, frozen=True)
-class DoneAwaitable:
-    """Simple awaitable that returns the provided value.
+async def maybe_awaitable(value: Union[Awaitable[R], R]) -> R:
```

Comment:
This seems much simpler, plus using the `isawaitable` call seems more inline with recommendations.

Reply:
It will mean that we add `maybe_awaitable` to the call stack, which is a bit sad.

Reply:
Ah, I see. I can revert this change if you'd prefer?